src/views/examples/Register.jsx

```diff
@@ -17,71 +18,209 @@ import {
-        }
-      });
+  // ./Password Toggle
+
+  // Form Validation
+
+  updateFirstName = firstName => {
+    this.setState({ firstName }, this.validateFirstName);
+  };
+
+  validateFirstName = () => {
+    const { firstName } = this.state;
+    let firstNameValid = true;
+    let errorMsg = { ...this.state.errorMsg };
+
+    // can't be blank/empty
+
+    if (firstName.length === 0) {
+      firstNameValid = false;
+      errorMsg.firstName = "Nome não pode ser vazio!";
     }
-  }
 
-  passwordMatcher = e => {
-    this.setState({ confirmPassword: e.target.value });
-    if (this.state.password === e.target.value) {
-      this.setState({
-        passMatch: {
-          text: "Senhas conferem!",
-          status: 1
-        }
-      });
-    } else {
-      this.setState({
-        passMatch: {
-          text: "Senhas não conferem!",
-          status: 2
-        }
-      });
+    this.setState({ firstNameValid, errorMsg }, this.validateForm);
+  };
+
+  updateSecondName = secondName => {
+    this.setState({ secondName }, this.validateSecondName);
+  };
+
+  validateSecondName = () => {
+    const { secondName } = this.state;
+    let secondNameValid = true;
+    let errorMsg = { ...this.state.errorMsg };
+
+    // can't be blank/empty
+
+    if (secondName.length === 0) {
+      secondNameValid = false;
+      errorMsg.secondName = "Sobrenome não pode ser vazio!";
     }
+
+    this.setState({ secondNameValid, errorMsg }, this.validateForm);
   };
 
-  handleSubmit = () => {
-    // Form validation
-    // API Call
+  updateCPF = cpf => {
+    this.setState({ cpf }, this.validateCPF);
   };
 
-  showHide(e) {
-    e.preventDefault();
-    e.stopPropagation();
+  validateCPF = () => {
+    const { cpf } = this.state;
+    let cpfValid = true;
+    let errorMsg = { ...this.state.errorMsg };
+
+    if (cpfMask(cpf).length > 0 && cpfMask(cpf).length < 14) {
+      cpfValid = false;
+      errorMsg.cpf = "Insira um CPF válido!";
+    } else if (cpfMask(cpf).length === 0) {
+      cpfValid = false;
+      errorMsg.cpf = "CPF não pode ser vazio!";
+    }
+
+    this.setState({ cpfValid, errorMsg }, this.validateForm);
+  };
+
+  updateMatriculation = matriculation => {
+    this.setState({ matriculation }, this.validateMatriculation);
+  };
+
+  validateMatriculation = () => {
+    const { matriculation } = this.state;
+    let matriculationValid = true;
+    let errorMsg = { ...this.state.errorMsg };
+
+    if (matriculation.length === 0) {
+      matriculationValid = false;
+      errorMsg.matriculation = "Matrícula não pode ser vazia!";
+    }
+
+    this.setState({ matriculationValid, errorMsg }, this.validateForm);
+  };
+
+  updateEmail = email => {
+    this.setState({ email }, this.validateEmail);
+  };
+
+  validateEmail = () => {
+    const { email } = this.state;
+    let emailValid = true;
+    let errorMsg = { ...this.state.errorMsg };
+
+    // checks for format _@_._
+    if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email)) {
```

Comment:
It would be nice to make a function to check the e-mail, at the line 39 you had used this Regex pattern.

Reply:
> It would be nice to make a function to check the e-mail, at the line 39 you had used this Regex pattern.

Check [this commit](https://github.com/ufv-crp/web-music/commit/a16e57469dfdf9e661dd892fd06993ac3f68c119)